Implement MoPub side suggestions

YueVungle opened the pull request:
According to MoPub side suggestions, I have made corresponding modifications.
The changes suggested by MoPub is in below document:
https://docs.google.com/document/d/1qtKXCZjkTzDRC-95LEJwVC6pS0aL53nMs-mJ2OaF07g/edit

crbinsf commented on Vungle/VungleBannerCustomEvent.m:
```diff
@@ -68,25 +62,25 @@ - (void)vungleAdDidLoad
         // VunglePlayAdOptionKeyUser
-        if ([[NSUserDefaults standardUserDefaults] objectForKey:kVungleUserId]) {
-            NSString *userID = [[NSUserDefaults standardUserDefaults] objectForKey:kVungleUserId];
+        if ([self.localExtras objectForKey:kVungleUserId]) {
```
@YueVungle - if we move to these, you may want to check the Sample Application and confirm how that should be updated for QA testing.

YueVungle replied:
I have checked the Sample App that is used by QA and added the logic about how the localExtras object will be updated by that application to pass in play options / settings. For the original NSUserDefaults accessing and saving logic, I didn't change them, they work as before, which will not affect Sample App test. I will send a PR for how the localExtras object will be updated of Sample App. :)

crbinsf commented on Vungle/VungleBannerCustomEvent.m:
```diff
@@ -133,6 +128,7 @@ - (void)vungleAdDidDisappear
 - (void)vungleAdWasTapped
 {
     MPLogInfo(@"Vungle video banner was tapped");
+    [self.delegate trackClick];
```
@YueVungle, discussing currently with @shiv-vungle - based on a comment from Dravya we do not want to call the `trackClick` method here. Unless otherwise noted, I would remove.

YueVungle replied:
Hi Clarke, according to Dravya’s comments, if we disable the automatically tracking and remove trackClick call, how MoPub knows user click.

crbinsf replied:
Per @shiv-vungle - he had a discussion with Matt Bennion and Dravya earlier today and confirmed that this line should be removed.

YueVungle replied:
Got it, remove this call.

YueVungle commented on Vungle/VungleBannerCustomEvent.m:
```diff
@@ -29,33 +29,32 @@ @interface VungleBannerCustomEvent () <VungleRouterDelegate>
 //    [[VungleRouter sharedRouter] completeBannerAdViewForPlacementID:self.placementId];
 }
 
+- (BOOL)enableAutomaticImpressionAndClickTracking
```
According to Dravya Nataraj’s comments “It would be better to disable auto tracking if we can track them in the adapters at the closest possible callbacks”. So I disable automatic tracking. :)

crbinsf replied:
> > @YueVungle thanks for sharing your changes. I've added two comments - one concerning the Sample App that is used by QA and how the `localExtras` object will be updated from that application to pass in play options / settings, and one concerning the `trackClick` method. Please let me know if you have any other follow up questions here. Thanks!
> 
> Hi Clarke, I have checked VungleInterstitialCustomEvent class, it also uses NSUserDefaults to get parameters to pass in play options / settings, which could be replaced by localExtras, do we need to modify it in this version or in 6.5.x version? Thanks so much. ^_^

@YueVungle thanks for following up on this - I believe we should use the same approach in both places - please use the `localExtras` approach in `VungleInterstitialCustomEvent` class as well (I've also commented in the test app PR regarding this). Thanks!

crbinsf replied:
Based on a conversation between @shiv-vungle and Dravya, we will not be implementing `trackClick` (mentioned below). As such, we should not implement `trackImpression` and keep the automatic impression and click tracking enabled for now. Matt Bennion says that proper click tracking support is on the roadmap, so we'll be able to revisit this at some point in the future.

YueVungle replied:
OK, remove override enableAutomaticImpressionAndClickTracking method. :)

YueVungle commented:
> @YueVungle thanks for sharing your changes. I've added two comments - one concerning the Sample App that is used by QA and how the `localExtras` object will be updated from that application to pass in play options / settings, and one concerning the `trackClick` method. Please let me know if you have any other follow up questions here. Thanks!

Hi Clarke, I have checked VungleInterstitialCustomEvent class, it also uses NSUserDefaults to get parameters to pass in play options / settings, which could be replaced by localExtras, do we need to modify it in this version or in 6.5.x version? Thanks so much. ^_^

crbinsf commented:
> > @YueVungle thanks for sharing your changes. I've added two comments - one concerning the Sample App that is used by QA and how the `localExtras` object will be updated from that application to pass in play options / settings, and one concerning the `trackClick` method. Please let me know if you have any other follow up questions here. Thanks!
> 
> Hi Clarke, I have checked VungleInterstitialCustomEvent class, it also uses NSUserDefaults to get parameters to pass in play options / settings, which could be replaced by localExtras, do we need to modify it in this version or in 6.5.x version? Thanks so much. ^_^

Hi @YueVungle - thanks so much for catching this - I believe we should update this so that the same approach of using `localExtras` is implemented for banners and `VungleInterstitialCustomEvent`. I've also updated the test app PR with a similar comment.

crbinsf commented on Vungle/VungleBannerCustomEvent.m:
```diff
@@ -118,6 +117,7 @@ - (void)vungleAdWillAppear
 
 - (void)vungleAdDidAppear {
     MPLogInfo(@"Vungle video banner did appear");
+    [self.delegate trackImpression];
```
Because we're not able to track clicks currently, we should also remove this and rely on the automatic impression and click tracking.

YueVungle replied:
Got it, remove this call. :)

YueVungle commented on Vungle/VungleRouter.m:
```diff
@@ -534,10 +534,6 @@ - (void)vungleWillCloseAdWithViewInfo:(VungleViewInfo *)info placementID:(NSStri
     }
 
     if(targetDelegate) {
-        if ([info.didDownload isEqual:@YES]) {
```
According to Clarke's comments, remove this logic.

YueVungle commented:
> > > @YueVungle thanks for sharing your changes. I've added two comments - one concerning the Sample App that is used by QA and how the `localExtras` object will be updated from that application to pass in play options / settings, and one concerning the `trackClick` method. Please let me know if you have any other follow up questions here. Thanks!
> > 
> > 
> > Hi Clarke, I have checked VungleInterstitialCustomEvent class, it also uses NSUserDefaults to get parameters to pass in play options / settings, which could be replaced by localExtras, do we need to modify it in this version or in 6.5.x version? Thanks so much. ^_^
> 
> Hi @YueVungle - thanks so much for catching this - I believe we should update this so that the same approach of using `localExtras` is implemented for banners and `VungleInterstitialCustomEvent`. I've also updated the test app PR with a similar comment.

@crbinsf  Hi Clarke , I have updated VungleInterstitialCustomEvent class, now it uses localExtras instead of NSUserDefaults to get parameters to pass in play options / settings. I also updated MoPub Adapter Test App to added the logic about how the localExtras object will be updated for interstitial by that application. For rewarded video, we don't need to do any change, because rewarded video uses VungleInstanceMediationSettings to pass in play options / settings.  ^_^

YueVungle commented on Vungle/VungleAdapterConfiguration.m:
```diff
@@ -41,7 +41,7 @@ + (void)updateInitializationParameters:(NSDictionary *)parameters {
 
 - (NSString *)adapterVersion {
-    return @"6.4.1";
+    return @"6.4.2.0";
```
Update adapter version to 6.4.2.0, which is corresponding to SDK 6.4.2
